cmd/beacon-chain/powchain/options.go

```diff
@@ -13,16 +19,53 @@ var log = logrus.WithField("prefix", "cmd-powchain")
 func FlagOptions(c *cli.Context) ([]powchain.Option, error) {
 	endpoints := parsePowchainEndpoints(c)
 	executionEndpoint := parseExecutionEndpoint(c)
+	jwtSecret, err := parseJWTSecretFromFile(c)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not read JWT secret file for authenticating execution API")
+	}
 	opts := []powchain.Option{
 		powchain.WithHttpEndpoints(endpoints),
 		powchain.WithEth1HeaderRequestLimit(c.Uint64(flags.Eth1HeaderReqLimit.Name)),
 	}
 	if executionEndpoint != "" {
 		opts = append(opts, powchain.WithExecutionEndpoint(executionEndpoint))
 	}
+	if len(jwtSecret) > 0 {
+		opts = append(opts, powchain.WithExecutionClientJWTSecret(jwtSecret))
+	}
 	return opts, nil
 }
 
+// Parses a JWT secret from a file path. This secret is required when connecting to execution nodes
+// over HTTP, and must be the same one used in Prysm and the execution node server Prysm is connecting to.
+// The engine API specification here https://github.com/ethereum/execution-apis/blob/main/src/engine/authentication.md
+// Explains how we should validate this secret and the format of the file a user can specify.
+//
+// The secret must be stored as a hex-encoded string within a file in the filesystem.
+// If the --jwt-secret flag is provided to Prysm, but the file cannot be read, or does not contain a hex-encoded
+// key of at least 256 bits, the client should treat this as an error and abort the startup.
+func parseJWTSecretFromFile(c *cli.Context) ([]byte, error) {
+	jwtSecretFile := c.String(flags.ExecutionJWTSecretFlag.Name)
+	if jwtSecretFile == "" {
+		return nil, nil
+	}
+	enc, err := file.ReadFileAsBytes(jwtSecretFile)
+	if err != nil {
+		return nil, err
+	}
+	if len(enc) == 0 {
+		return nil, fmt.Errorf("provided JWT secret in file %s cannot be empty", jwtSecretFile)
+	}
+	secret, err := hex.DecodeString(strings.TrimPrefix(string(enc), "0x"))
```

Comment:
geth's hexutil library actually errors if a string is not 0x prefixed. We want to allow some flexibility here and not be so harsh on users